### PR TITLE
Fix: Implemented Conditional GET using etags

### PR DESCRIPTION
**Pull request by CalebKussmaul**
On non-range requests, kitura StaticFileServer will never send HTTP 304 NOT MODIFIED response, even if client already has the latest version of a resource. It will send the entire file instead, which is a waste of resources. Additionally, a documentation error regarding the max-age header parameter was fixed. 

## Description

A check was added to see if the file's Etag matched the request's corresponding "If-None-Match" parameter. If they are equal, the server will respond with HTTP 304 NOT MODIFIED. Otherwise it will still send the file.

## Motivation and Context

See issue #1332. In some use cases, a browser may request the same resource on many different pages, or multiple times for a number of other reasons. Using the max-age parameter is inadequate because it does not allow the developer to update a resource at any time without clients continuing to use an outdated version of a resource. I created this because I need my server to be as efficient as possible, and under my use case this change results in a significant resource savings. 

## How Has This Been Tested?

I have tested the code on a personal project (on a mac using a browser on the same computer). It should not affect other areas of the code. 

## Checklist:

- [x] If applicable, I have updated the documentation accordingly.
- [ ] If applicable, I have added tests to cover my changes.

### CODE REVIEW

**Comment by ianpartridge** [comment]
Hi @CalebKussmaul! Are you able to update this PR? We'd like to merge it :)

**Comment by CalebKussmaul** [comment]
@ianpartridge It should be all set now, thank you!

**Comment by ianpartridge** [comment]
@djones6 can you take another look please :)

File: Sources/Kitura/staticFileServer/FileServer.swift [hunk 1]
```diff
@@ -187,9 +187,15 @@ extension StaticFileServer {
                         // Send only headers
                         _ = response.send(status: .OK)
                     } else {
-                        // Send the entire file
-                        try response.send(fileName: filePath)
-                        response.statusCode = .OK
+                        if let etag = request.headers["If-None-Match"],
+                          etag == CacheRelatedHeadersSetter.calculateETag(from: fileAttributes) {
+                            response.statusCode = .notModified
+                            response.send("")
```
Comment (djones6) [inline Sources/Kitura/staticFileServer/FileServer.swift]: I don't think this empty `send("")` is necessary - it'll do some work under the covers, but result in no data being appended to the response buffer.  Setting the `statusCode` is sufficient.

One of the outcomes of invoking `send()` is that it sets `response.state.invokedSend` to indicate that a response has/is being sent.  However, we only send a default response if `invokedSend` is false _and_ `statusCode` is `.unknown`.

**Comment by ianpartridge** [comment]
Thanks @CalebKussmaul!  @djones6 - please rereview.

**Pull request merged**